hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -442,8 +443,21 @@ public static String addOrIncrementIndexInName(String name) {
         return name + '-' + index;
     }
 
-    public static String sanitizeLoggerNamePart(String name) {
-        return name.replace('.', '_');
+    public static ILogger prefixedLogger(ILogger logger, String prefix) {
+        return new PrefixedLogger(logger, prefix);
+    }
+
+    public static String prefix(String jobName, long jobId, String vertexName) {
+        return prefix(jobName, jobId, vertexName, null);
+    }
+
+    public static String prefix(String jobName, long jobId, String vertexName, int processorIndex) {
+        return prefix(jobName, jobId, vertexName, "procIndex:" + processorIndex);
+    }
+
+    public static String prefix(String jobName, long jobId, String vertexName, String identity) {
+        String jobIdentification = StringUtil.isNullOrEmptyAfterTrim(jobName) ? idToString(jobId) : jobName.trim();
```

Comment:
The `jobName` is an identifier, don't trim it. If you name your job `myjob ` (with the trailing space), you need that trailing space when you do `JetInstance.getJob("myjob ")`. Even an empty job name is a valid job name.
```suggestion
        String jobIdentification = jobName != null ? jobName : idToString(jobId);
```